src/main/scala/viper/gobra/frontend/info/implementation/property/Assignability.scala

```diff
@@ -159,6 +166,41 @@ trait Assignability extends BaseProperty { this: TypeInfoImpl =>
     case (l, t) => failedProp(s"cannot assign literal $l to $t")
   }
 
+  def assignWithinBounds: Property[(Type, PExpression)] = createFlatProperty[(Type, PExpression)] {
+    case (typ, expr) => s"constant expression $expr overflows $typ"
+  } {
+    case (typ, expr) =>
+      val constVal = intConstantEval(expr)
+      constVal.isEmpty || intValInBounds(constVal.get, typ)
+  }
+
+  private def intValInBounds(value: BigInt, typ: Type): Boolean =
+    underlyingType(typ) match {
+      case IntT(t) => t match {
+        case UntypedConst => true
+        case x =>
+          val bounds = getBoundsOfIntKind(x)
+          bounds.lower <= value && value <= bounds.upper
+      }
+
+      case _ => ???
```

Comment:
This should throw an error, right? I would use ??? only if something is supposed to be there eventually.

Reply:
Yes, you are correct

Reply:
Fixed in bceb60f